src/test/java/test/io/ticktok/client/tick/HttpTickerPolicyTest.java

```diff
@@ -0,0 +1,185 @@
+        register("no-ticks", tickCount::incrementAndGet);
+        assertNoTicks(tickCount);
+    }
+
+    private void assertNoTicks(AtomicInteger tickCount) throws InterruptedException {
+        sleep(SECOND);
+        assertThat("Ticks were consumed", tickCount.get(), is(0));
+    }
+
+    private void register(String id, TickConsumer consumer) {
+        tickListener.forChannel(httpChannel(id)).register(consumer);
+    }
+
+    private TickChannel httpChannel(String id) {
+        Map<String, String> details = new HashMap<>();
+        details.put("url", format("http://localhost:%s/%s/pop", PORT, id));
+        return TickChannel.builder().type("http").details(details).build();
+    }
+
+    @Test
+    void ignoreOnRequestError() throws Exception {
+        server = new TicktokServer().onPopFailWith(500);
+        assertNoTicksWereConsumed();
+        assertTrue(server.popped);
+    }
+
+    @Test
+    void invokeOnTick() throws Exception {
+        server = new TicktokServer().tickOn("tick-me");
+        PacedTickCounter tickCounter = new PacedTickCounter(2);
+        register("tick-me", tickCounter::countDown);
+        tickCounter.assertNoTicks();
+        assertTimeoutPreemptively(ofSeconds(2), tickCounter::await, "No ticks");
+    }
+
+    @Test
+    void ignoreWhenThereNoTicksToPop() throws Exception {
+        server = new TicktokServer();
+        assertNoTicksWereConsumed();
+    }
+
+    @Test
+    void disconnectAllClocks() throws Exception {
+        server = new TicktokServer();
+        AtomicInteger tickCount = new AtomicInteger();
+        register("c1", tickCount::incrementAndGet);
+        register("c2", tickCount::incrementAndGet);
+        tickListener.disconnect();
+        sleep(100);
+        server.tickOn("c1", "c2");
+        assertNoTicks(tickCount);
+    }
+
+    @Test
+    void allowToRegisterAfterDisconnect() throws Exception {
+        server = new TicktokServer().tickOn("c1");
+        tickListener.disconnect();
+        CountDownLatch tickConsumed = new CountDownLatch(1);
+        register("c1", tickConsumed::countDown);
+        assertTimeoutPreemptively(ofSeconds(1), (Executable) tickConsumed::await);
+    }
+
+    @Test
+    void replaceCallbackForAGivenClock() throws Exception {
+        server = new TicktokServer();
+        CountDownLatch countForOriginal = new CountDownLatch(1);
+        CountDownLatch countForNew = new CountDownLatch(1);
+        register("c1", countForOriginal::countDown);
+        register("c1", countForNew::countDown);
+        server.tickOn("c1");
+        assertTimeoutPreemptively(ofSeconds(1), (Executable) countForNew::await);
+        assertThat("original consumer should not get ticks", countForOriginal.getCount(), is(1L));
+    }
+
+    @AfterEach
+    void stopServer() {
+        tickListener.disconnect();
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    static class TicktokServer extends BlinkServer {
```

Comment:
why need to extend Blink. you should avoid extending 3th parties libs. even if you wrote it.
you dont want to relay on behavior you dont know,

Reply:
never hear of this rule
I don't see the problem here 
and I certainly rely on a known api when extending a class 
nothing is unknown here

Reply:
ill give an example (actually a real one that i experienced).
imagine you have in your code  a class that extends a map, simple java map.
your code is bounded to maps behavior, which you may want or not.
even java has a know open api.

in fact we had a lot of bugs, and keep in mind you bound your class behavior to some other api behavior and it can change without you have any control over.

hope it clarify for you my stand

Reply:
Actually extending many classes in Java is the standard, think of: JButton, Exceptions, etc.
I do expect code to inherit blink. in fact using the {{ }} is actually extending blink implicitly